libs/datamodel/core/tests/attributes/default_composite_positive.rs

```diff
@@ -0,0 +1,197 @@
+use crate::common::*;
+use bigdecimal::{BigDecimal, FromPrimitive};
+use chrono::DateTime;
+use datamodel::{DefaultValue, ScalarType, ValueGenerator};
+use dml::prisma_value::PrismaValue;
+
+#[test]
+fn should_set_default_for_all_scalar_types() {
+    let dml = r#"
+    datasource db {
+        provider = "mongodb"
+        url = "mongodb://"
+    }
+
+    type Model {
+        int Int @default(3)
+        float Float @default(3.20)
+        string String @default("String")
+        boolean Boolean @default(false)
+        dateTime DateTime @default("2019-06-17T14:20:57Z")
+        bytes    Bytes @default("aGVsbG8gd29ybGQ=")
+        json     Json  @default("{ \"a\": [\"b\"] }")
+        decimal  Decimal  @default("121.10299000124800000001")
+    }
+    "#;
+
+    let datamodel = parse(dml);
+    let user_model = datamodel.assert_has_composite_type("Model");
+    user_model
+        .assert_has_scalar_field("int")
+        .assert_base_type(&ScalarType::Int)
+        .assert_default_value(DefaultValue::new_single(PrismaValue::Int(3)));
+    user_model
+        .assert_has_scalar_field("float")
+        .assert_base_type(&ScalarType::Float)
+        .assert_default_value(DefaultValue::new_single(PrismaValue::Float(
+            BigDecimal::from_f64(3.20).unwrap(),
+        )));
+
+    user_model
+        .assert_has_scalar_field("string")
+        .assert_base_type(&ScalarType::String)
+        .assert_default_value(DefaultValue::new_single(PrismaValue::String(String::from("String"))));
+    user_model
+        .assert_has_scalar_field("boolean")
+        .assert_base_type(&ScalarType::Boolean)
+        .assert_default_value(DefaultValue::new_single(PrismaValue::Boolean(false)));
+    user_model
+        .assert_has_scalar_field("dateTime")
+        .assert_base_type(&ScalarType::DateTime)
+        .assert_default_value(DefaultValue::new_single(PrismaValue::DateTime(
+            DateTime::parse_from_rfc3339("2019-06-17T14:20:57Z").unwrap(),
+        )));
+    user_model
+        .assert_has_scalar_field("bytes")
+        .assert_base_type(&ScalarType::Bytes)
+        .assert_default_value(DefaultValue::new_single(PrismaValue::Bytes(b"hello world".to_vec())));
+    user_model
+        .assert_has_scalar_field("json")
+        .assert_base_type(&ScalarType::Json)
+        .assert_default_value(DefaultValue::new_single(PrismaValue::Json(
+            r#"{ "a": ["b"] }"#.to_owned(),
+        )));
+    user_model
+        .assert_has_scalar_field("decimal")
+        .assert_base_type(&ScalarType::Decimal)
+        .assert_default_value(DefaultValue::new_single(PrismaValue::Float(
+            r#"121.10299000124800000001"#.parse().unwrap(),
+        )));
+}
+
+// TODO: Remove ignore when enums are supported
```

Comment:
I don't think it's a good idea to include tests with `TODO`. We have quite a lot of them already that's been there for a few years. We can write them when we actually decide on implementing these.

Reply:
I have a follow-up PR that removes them: https://github.com/prisma/prisma-engines/pull/2608